aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -163,7 +163,7 @@ static void sign_kernel(TensorIterator& iter){
   if(iter.dtype() == ScalarType::Bool){
       cpu_kernel(iter, [=](bool x) -> bool { return x; });
   } else {
-    AT_DISPATCH_ALL_TYPES_AND2(kBFloat16, ScalarType::Half, iter.dtype(), "sign_cpu", [&]() {
+    AT_DISPATCH_ALL_TYPES_AND2(kBFloat16, kHalf, iter.dtype(), "sign_cpu", [&]() {
```

Comment:
I'm actually creating a test suite that will make it easy to test this soon. It should probably wait for https://github.com/pytorch/pytorch/pull/41249.